File: model/php/srv.php
<?php
/*
* Hello World server
* Binds REP socket to tcp://*:5555
* Expects "Hello" from client, replies with "World"
* @author Ian Barber <ian(dot)barber(at)gmail(dot)com>
*/
echo "Starting\n";
$context = new ZMQContext(1);

// Socket to talk to clients
$responder = new ZMQSocket($context, ZMQ::SOCKET_REP);
$responder->bind("tcp://*:5555");

echo "Bound and waiting\n";

while (true) {
    process($responder);
}

function process(ZMQSocket &$responder) {

    // Wait for next request from client
    $request = $responder->recv();
//    printf ("Received request: [%s]\n", $request);

    // Do some 'work'
    $data = json_decode($request);

    $metricsSimple = [];
    foreach ($data->metrics as $m) {
        $metricsSimple[$m->k] = $m->v;
    }

    $prob = json_encode(['probability' => (
              ($metricsSimple['age'] * .4)
            + ($metricsSimple['score'] * .8)
            + ($metricsSimple['timeOnHome'] * .04)
            + ($metricsSimple['timeOnBankSearch'] * .1)
            + ($metricsSimple['timeOnCCSearch'] * .2)
            + ($metricsSimple['viewedCCOfferAmex'] * .4)
            + ($metricsSimple['viewedCCOfferCapOne432'] * .4)
        )]);

    // Send reply back to client
    $responder->send($prob);

}

File: model/go/srv.go
package main

import (
	zmq "github.com/pebbe/zmq3"
	"fmt"
	"encoding/json"
)

type Metric struct {
	K string `json:"k"`
	V int `json:"v"`
}

type Request struct {
	Metrics []Metric
}

type Response struct {
	Probability float32 `json:"probability"`
}

func main() {
	responder, _ := zmq.NewSocket(zmq.REP)
	defer responder.Close()
	responder.Bind("tcp://*:5555")

	for {
		msg, _ := responder.Recv(0)
//		fmt.Println("Received ", msg)

		var request Request
		err := json.Unmarshal([]byte(msg), &request)
		if err != nil {
			fmt.Println("error:", err)
		}
//		fmt.Printf("%+v\n", request)

		// turn metrics into lookup map
		var lookup map[string]float32
		lookup = make(map[string]float32)
		for _, value := range request.Metrics {
			lookup[value.K] = float32(value.V);
		}

//		fmt.Printf("%+v\n", lookup)

		p := ((lookup["age"] * 0.4) + (lookup["score"] * 0.8) + (lookup["timeOnHome"] * 0.04) + (lookup["timeOnBankSearch"] * 0.1) + (lookup["timeOnCCSearch"] * 0.2) + (lookup["viewedCCOfferAmex"] * 0.4) + (lookup["viewedCCOfferCapOne432"] * 0.4))
		prob := Response{Probability: p}

//		fmt.Printf("%+v\n", prob)

		b, err := json.Marshal(prob)
		if err != nil {
			fmt.Println("error:", err)
		}

		// send it back!
		responder.Send(string(b), 0)

//		responder.Send("hi", 0)
	}

}
